File: module_main/src/main/java/com/frame/mvp/main/FragmentAdapter.java
package com.frame.mvp.main;

import android.view.View;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewpager.widget.PagerAdapter;

public class FragmentAdapter extends PagerAdapter {
    @Override
    public int getCount() {
        return 0;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return false;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return super.getPageTitle(position);
    }
}

File: module_main/src/main/java/com/frame/mvp/main/FragmentAdapter.kt
package com.frame.mvp.main

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.frame.mvp.common.bean.FragmentBean

class FragmentAdapter(fm: FragmentManager, private val fragmentList: MutableList<FragmentBean>) :
    FragmentPagerAdapter(fm) {

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getItem(position: Int): Fragment? {
        return fragmentList[position].fragment
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentList[position].title
    }
}
